### ISSUE TITLE
[REQ] Watt Gain Setting available via Tile and export only real Wattage to Strava 

### ISSUE THREAD

**Issue by Bepo7012** [cagnulein/qdomyos-zwift#1715]
Hi @cagnulein 

Is it maybe possible to make the Watt Gain Setting available and decrease/increase it via a new Tile? I would like to have this one because it would be the easiest way to simulate a E-Bike with QZ. The Reason i need this is, because i want to ride my Video Recordings i made with my E-Bike and because of my Injury i'm not able to apply the high Wattage needed to ride this Tracks without "Support" :-) Setting the Watt Gain permanently is not a good solution because then it is too high on leveled parts of the Track or on Parts with decreasing Inclination.

Additionally it would be nice to export only the real pedalled Wattage to Strava for seeing and comparing the real Training Training Results (Bosch's FlowApp is doing this quite nice, they only export the real applied Wattage to Strava).

What do you think about such a suggestion?

**Comment by cagnulein** [comment]
@Bepo7012 I would love the idea!

I also saw that in the fit file we can handle these fields

```
{ 0, 0, "ebike_travel_range", "km", (FIT_FLOAT64)1, (FIT_FLOAT64)0, 0, 0, 117, 132, false, Profile::Type::Uint16 },
{ 0, 0, "ebike_battery_level", "percent", (FIT_FLOAT64)1, (FIT_FLOAT64)0, 0, 0, 118, 2, false, Profile::Type::Uint8 },
{ 0, 0, "ebike_assist_mode", "depends on sensor", (FIT_FLOAT64)1, (FIT_FLOAT64)0, 0, 0, 119, 2, false, Profile::Type::Uint8 },
{ 0, 0, "ebike_assist_level_percent", "percent", (FIT_FLOAT64)1, (FIT_FLOAT64)0, 0, 0, 120, 2, false, Profile::Type::Uint8 },
```

except for the first 2, i guess assist mode and assist level percent could be handle by this mod. For example if the watt gain is 1 the assist mode is 0. when watt gain is higher than 1, assist mode will be 1 and the level percent can be scaled accordingly to the watt gain.

What do you think?

**Comment by Bepo7012** [comment]
@cagnulein Whereas it is nice that there are Ebikes who record the Informations i don't think it is a needed way to read the Information: First of all, there is mostly a .gpx where this Information is in generally not available, second, not all eBikes will record the Data (i.E. i didn't find any Data in Bosch's .fit Files) and third, every rider and even more every day can be different. Climbs which are normally undoable the one day can be done on another day easily. So making it adjustable "on the fly" would be the better Option i think.

**Comment by cagnulein** [comment]
no i guess we misunderstood! i was thinking that qz will write this fields
in the fit file accordingly to the watt gain, not the opposite :)

Il giorno gio 19 ott 2023 alle 09:19 Bepo7012 ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
yes it's like the concept of the "difficulty" on zwift. You can do already actually using the plus and minus button on the target resistance tile. Maybe it could be enough already! 

But I guess you were thinking about a hybrid tile that could contain both, isn't it?

**Comment by Bepo7012** [comment]
I didn't know that the target resistance tile has any functionality on riding gpx/Video Tracks but i will give it a try for sure when i'm back home. But a hybrid Tile as suggested would be a great thing because i think it would react quite the same as the Support Modes on a eBike. Not completely because the new eBikes add more and more dynamic modes which are no longer giving a fixed Support by Percent to the Wattage pedaled but reacts on the way the Rider is pedaling. Having a one touch possibility to influence inclination and watt gain should be a great thing i think.

And exporting the real Wattage to Strava would make the whole Thing complete since the exported Data show's the "real Life" :-)

**Comment by cagnulein** [comment]
so summarizing:
- a new tile that combines watt gain (+/- 0.10 at each press or customizable?) and one (or customizable) resistance levels. 5 levels of assistance (or more?)
- qz will write (and shows?) the real wattage

**Comment by Bepo7012** [comment]
Sounds perfect. 5 assistance Levels should be enough, it's the same as most of the eBike Manufacturer have. Although it would blow up the Options Screen again i would prefer to set the Watt Gain (fixed) and the resistance Levels (+/- on the original calculated Value) for each Assistance Level. +/- 0.1 on the watt gain would not reflect the Support from an eBike. I.E. Bosch Support Modes are currently (i added the dynamic Modes just for completeness):

- Eco: 60%
- Tour: 140%
- Tour+: 60%-340%
- eMTB: 140%-340%
- Turbo: 340%

This Behaviour would not be manageable with a +/- watt gain on each Press.

And yes, writing the real wattage is necesserily, showing the real wattage would be great too since this is the truth ;-)

And maybe deactivate the whole Feature for Zwift for not opening the Doors for Cheaters?

And the next coffee's are my shout ;-)

**Comment by cagnulein** [comment]
> Sounds perfect. 5 assistance Levels should be enough, it's the same as most of the eBike Manufacturer have. Although it would blow up the Options Screen again i would prefer to set the Watt Gain (fixed) and the resistance Levels (+/- on the original calculated Value) for each Assistance Level. +/- 0.1 on the watt gain would not reflect the Support from an eBike. I.E. Bosch Support Modes are currently (i added the dynamic Modes just for completeness):

Ok I will do a separate page for this since having 10 other new settings for everyone could be a mess :)

> This Behaviour would not be manageable with a +/- watt gain on each Press.

yes got it ;)

> And yes, writing the real wattage is necesserily, showing the real wattage would be great too since this is the truth ;-)

this one will be challenging, beucase now QZ has a wattage only metrics... I guess I would have to separate the raw wattage from the calculated wattage

> And maybe deactivate the whole Feature for Zwift for not opening the Doors for Cheaters?

I'm not worried about this, everyone is free to do whatever they wants with their bikes (this is my opinion of course).

**Comment by cagnulein** [comment]
btw are you able to do tests with the windows build that github is providing us? this will be free me to create an IOS build everytime we need to test something about this :)

**Comment by Bepo7012** [comment]
Yes i can test it with Windows (and a maximum flickering Video playback which isn't a problem for this case :-) ) in about 2 Weeks when i have finished the torture here :-D

**Comment by cagnulein** [comment]
> * Tour+: 60%-340%
> * eMTB: 140%-340%

how do they work in reality? they are based on cadence?

**Comment by Bepo7012** [comment]
I don't know exactly how they work, don't know if Bosch made the exact Specifications public. As far as i know and what i experience on the Bike is, that they have to combine cadence and pedaled wattage. I.E: if you ride on a leveled Road you get less Support. If then comes a climb and you have to apply more watts to hold the cadence you get a bit more support. if the inclination gets higher and cadence goes down you get more and more support until your applied wattage is enough to hold or increase cadence. Also if the applied wattage goes down because you get tired support is increased. When support is then high enough to increase cadence support is be lowered again.

Additionally (especally in eMTB Mode) there is a Kind of Boost you get when applied wattage is jumping up. So if you have to climb up a high Inclination and you abrutly kick in the Pedals you get an additionally high Boost to overcome the Obstacle.

But as mentioned i only added both for completeness, don't think that there is a need to implement this active Modes in QZ (there is no need to overcome obstacles with the Ergometer :-D )

**Comment by Bepo7012** [comment]
Hi Roberto

First of all, a happy new year to you and your Familiy :-) 

I know you have quite a lot ToDo's currently, but is there any expected Timeline on this?

Bernhard

**Comment by cagnulein** [comment]
hi @Bepo7012 happy new year too!

i'm currently overwhelmed with requests and this one is not a small one since it involves changing a lot of code lines. 

i don't think i will be able to do before this spring i guess

aren't you able to start a PR for this?
